Fixed missing required info in resource summary

shashigharti opened the pull request:
Fixed missing info of required property in resource summary
Added tests

- fixes #1161

---

Please make sure that all the checks pass. Please add here any additional information regarding this pull request. It's highly recommended that you link this PR to an issue (please create one if it doesn't exist for this PR)

shashigharti committed:
Fixed missing required info in resource summary

Fixed missing info of required property in resource summary
Added tests

shashigharti commented:
@roll this is ready to review!

roll reviewed (changes requested):
Hi @shashigharti,

I think the initial issue is invalid as there is no `field.required` property in the standards. It should be provided as `field.constraint.required`.

Can you please:
- keep all the work you done in this PR it's anyway useful changes
- but switch the required setting in tests to a proper one (`field.constraints.required`)
- WDYT if we emit a validation error if `field.required` is provided as we did for `resource.fields` etc?

roll commented on frictionless/schema/schema.py:
```diff
@@ -262,6 +262,10 @@ def to_excel_template(self, path: str):
 
     def to_summary(self) -> str:
         """Summary of the schema in table format"""
+        # Setting required property of the field which is included as custom fields
```
So we need to remove these 4 lines

shashigharti replied:
Now I understand, I will remove these lines. I should have checked the standards. Sure adding "validation error" would be better. I will add that.

shashigharti committed:
Revised code

Removed extra code to check if required field exists
Added error message to notify invalid Field property required
Added new tests

shashigharti commented:
Thank you @roll . I have made the changes and it is ready to review!

roll commented on frictionless/schema/field.py:
```diff
@@ -218,6 +218,13 @@ def metadata_import(cls, descriptor):
         return super().metadata_import(descriptor)
 
     def metadata_validate(self):
+        invalid_fields = {"required": "field->constraints"}
```
@shashigharti 
Looks good but I slightly simplified these checks during the v5 work. Can you please follow this pattern?

> e.g. resource.py

```
        # Custom
        for name in ["missingValues", "fields"]:
            if name in self.custom:
                note = f'"{name}" should be set as "schema.{name}"'
                yield errors.ResourceError(note=note)
```

shashigharti replied:
Thank you!. I had checked that but got confused. I will fix that.

shashigharti committed:
Changed code to align with v5 changes

shashigharti commented:
@roll thank you! I have made the changes. It is ready for the review!

roll reviewed (approved):
Thanks!
